supervisor/updater.py

```diff
@@ -56,17 +56,26 @@ def __init__(self, coresys: CoreSys) -> None:
 
     async def load(self) -> None:
         """Update internal data."""
-        # If there's no connectivity, delay initial version fetch
-        if not self.sys_supervisor.connectivity:
-            self._connectivity_listener = self.sys_bus.register_event(
-                BusEvent.SUPERVISOR_CONNECTIVITY_CHANGE, self._check_connectivity
+        # Delay loading data by default so JobCondition.OS_SUPPORTED works.
+        # Use HAOS unrestricted as indicator as this is what we need to evaluate
+        # if the operating system version is supported.
+        if self.sys_os.board and self.version_hassos_unrestricted is None:
+            _LOGGER.info(
+                "No OS update information found, force refreshing updater information"
             )
-            return
-
-        await self.reload()
+            await self.reload()
 
     async def reload(self) -> None:
         """Update internal data."""
+        # If there's no connectivity, delay initial version fetch
+        if not self.sys_supervisor.connectivity:
+            if not self._connectivity_listener:
+                self._connectivity_listener = self.sys_bus.register_event(
+                    BusEvent.SUPERVISOR_CONNECTIVITY_CHANGE, self._check_connectivity
+                )
+            _LOGGER.info("No Supervisor connectivity, delaying version fetch")
+            return
+
```

Comment:
So when this code was in `load` it would only delay the initial version fetch since `load` was only called once.

Now that this code is in `reload` it will happen repeatedly. We call `reload` on a schedule from a task. Any time that is called and the internet is down it will delay that until the internet comes back up. Whereas in the past it would've simply skipped that call to `reload` and tried again next time.

I think its ok like this but wanted to clarify the new behavior. Because this is a change in behavior, that comment is a little misleading. It's not just the initial fetch that can get delayed anymore, its all of them.